x-pack/test/ui_capabilities/security_and_spaces/scenarios.ts

```diff
@@ -0,0 +1,520 @@
+          privileges: ['all'],
+        },
+      ],
+    },
+    kibana: {
+      global: {
+        minimum: ['all'],
+      },
+    },
+  },
+};
+
+const DualPrivilegesRead: User = {
+  username: 'dual_privileges_read',
+  fullName: 'dual_privileges_read',
+  password: 'dual_privileges_read-password',
+  role: {
+    name: 'dual_privileges_read_role',
+    elasticsearch: {
+      indices: [
+        {
+          names: ['.kibana*'],
+          privileges: ['read'],
+        },
+      ],
+    },
+    kibana: {
+      global: {
+        minimum: ['read'],
+      },
+    },
+  },
+};
+
+const GlobalAll: User = {
+  username: 'global_all',
+  fullName: 'global_all',
+  password: 'global_all-password',
+  role: {
+    name: 'global_all_role',
+    kibana: {
+      global: {
+        minimum: ['all'],
+      },
+    },
+  },
+};
+
+const GlobalRead: User = {
+  username: 'global_read',
+  fullName: 'global_read',
+  password: 'global_read-password',
+  role: {
+    name: 'global_read_role',
+    kibana: {
+      global: {
+        minimum: ['read'],
+      },
+    },
+  },
+};
+
+const EverythingSpaceAll: User = {
+  username: 'everything_space_all',
+  fullName: 'everything_space_all',
+  password: 'everything_space_all-password',
+  role: {
+    name: 'everything_space_all_role',
+    kibana: {
+      space: {
+        everything_space: {
+          minimum: ['all'],
+        },
+      },
+    },
+  },
+};
+
+const EverythingSpaceRead: User = {
+  username: 'everything_space_read',
+  fullName: 'everything_space_read',
+  password: 'everything_space_read-password',
+  role: {
+    name: 'everything_space_read_role',
+    kibana: {
+      space: {
+        everything_space: {
+          minimum: ['read'],
+        },
+      },
+    },
+  },
+};
+
+const NothingSpaceAll: User = {
+  username: 'nothing_space_all',
+  fullName: 'nothing_space_all',
+  password: 'nothing_space_all-password',
+  role: {
+    name: 'nothing_space_all_role',
+    kibana: {
+      space: {
+        nothing_space: {
+          minimum: ['all'],
+        },
+      },
+    },
+  },
+};
+
+const NothingSpaceRead: User = {
+  username: 'nothing_space_read',
+  fullName: 'nothing_space_read',
+  password: 'nothing_space_read-password',
+  role: {
+    name: 'nothing_space_read_role',
+    kibana: {
+      space: {
+        nothing_space: {
+          minimum: ['read'],
+        },
+      },
+    },
+  },
+};
+
+export const Users: User[] = [
+  NoKibanaPrivileges,
+  Superuser,
+  LegacyAll,
+  LegacyRead,
+  DualPrivilegesAll,
+  DualPrivilegesRead,
+  GlobalAll,
+  GlobalRead,
+  EverythingSpaceAll,
+  EverythingSpaceRead,
+  NothingSpaceAll,
+  NothingSpaceRead,
+];
+
+// these are the spaces that we care about
+const EverythingSpace: Space = {
+  id: 'everything_space',
+  name: 'everything_space',
+  disabledFeatures: [],
+};
+
+const NothingSpace: Space = {
+  id: 'nothing_space',
+  name: 'nothing_space',
+  disabledFeatures: [
+    'advancedSettings',
```

Comment:
Any chance we can populate this and [NavLinks](https://github.com/elastic/kibana/pull/27414/files#diff-f2e89f65dacbaf604cf9e5542e362879R7) from the same data set? It'd be one less place to update when adding a new application. It's ok to say "no" or "absolutely not, that's a horrible idea" 😄